WordPress/openverse
# Rename ContentProvider to ContentSource

## Fixes



Fixes #4346 by @krysal

## Description


This PR renames the `ContentProvider` model to `ContentSource` to correctly reflect what the model is used for.

Interestingly, we had an inconsistency in included/excluded source query. The included sources query (`?source=<source_name>`) was querying _sources_ in ES, but the excluded sources query (`?excluded_source=<source_name>) was querying _providers_.
This PR changes the excluded sources query to exclude _sources_, not _providers_, as I think this is the correct way, and the documented way. I don't think it will affect the results much, and would like to hear reviewer opinions.

## Testing Instructions

The CI should pass.
The API should behave as usual:
- http://localhost:50280/v1/images/stats - should return all image sources (which are synonymous with providers in the sample data)
- http://localhost:50280/v1/images?source=wikimedia should return only the Wikimedia images
- http://localhost:50280/v1/images?excluded_source=flickr should return images from sources other than Flickr

Also try setting "Hide content" to true for one of the sources in the admin, and then check that `/stats` endpoint does not contain this source anymore. Also, `?source=<hidden_source>` should not return any results.

## Checklist


      PRs) or the media properties generator (`just catalog/generate-docs media-props`
      for the catalog or `just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
